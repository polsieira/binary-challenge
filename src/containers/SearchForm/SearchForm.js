import React, { Component } from 'react';
import { connect } from 'react-redux';
import './SearchForm.scss';
import { bindActionCreators } from 'redux';
import Typography from '@material-ui/core/Typography';
import Slider from '@material-ui/core/Slider';
import SearchIcon from '@material-ui/icons/Search';
import { addFilters } from '../../actions';
import PropTypes from 'prop-types';

export class SearchForm extends Component {
  constructor() {
    super()
    this.state = {
      distance: 10,
      gravity: 1,
      year: 365,
    }
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: parseInt(event.target.innerText)
    });
  };

  handleClick = () => {
    const { addFilters } = this.props;
    addFilters(this.state);
  }

  render() {
    return (
      <div className="SearchForm">
        <form className="search-form">
          <div className="input">
            <Typography id="range-slider" gutterBottom>
              Distance In Lightyears
          </Typography>
            <Slider
              id="distance"
              name="distance"
              onChange={this.handleChange}
              valueLabelDisplay="auto"
              aria-labelledby="range-slider"
              min={0}
              max={100}
            />
          </div>
          <div className="input">
            <Typography id="range-slider" gutterBottom>
              Comparative Gravity
          </Typography>
            <Slider
              id="gravity"
              name="gravity"
              onChange={this.handleChange}
              valueLabelDisplay="auto"
              aria-labelledby="range-slider"
              min={0.0001}
              max={100}
            />
          </div>
          <div className="input">
            <Typography onChange={this.handleChange} id="range-slider" gutterBottom>
              Length of Year In Days
          </Typography>
            <Slider
              id="year"
              name="year"
              onChange={this.handleChange}
              valueLabelDisplay="auto"
              aria-labelledby="range-slider"
              min={1}
              max={2000}
            />
          </div>
          <button type="button" className="search-button" onClick={this.handleClick}>
            <SearchIcon className="search-icon" />
          </button>
        </form>
      </div>
    )
  }

}

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    addFilters
  }, dispatch)
)

export default connect(null, mapDispatchToProps)(SearchForm);

SearchForm.propTypes = {
  addFilters: PropTypes.func
}