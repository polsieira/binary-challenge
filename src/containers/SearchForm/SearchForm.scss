.SearchForm {

  .search-form {
    background: white;
    bottom: 50px;
    box-sizing: border-box;
    border-radius: 5px;
    color: black;
    display: flex;
    justify-content: space-between;
    margin: 0px auto;
    position: relative;
    width: 90%;

    .input {
      align-content: flex-end;
      border-radius: 4px;
      display: flex;
      font-size: 10px;
      flex-direction: column;
      justify-content: flex-end;
      margin: 20px;
      padding: 10px;
      width: 180px;

      .MuiTypography-body1 {
        font-size: 16px;
      }

    }
    .search-button {
      background-color: #3F51B5;
      border: #3F51B5 solid 2px;
      border-top-right-radius: 5px;
      border-bottom-right-radius: 5px;
      width: 80px;

      &:hover {
        cursor: pointer;
        opacity: 0.8;
      }

      &:focus {
        outline: none;
      }
    }

    .search-icon {
      cursor: pointer;
      pointer-events: none;

    }
  }
}