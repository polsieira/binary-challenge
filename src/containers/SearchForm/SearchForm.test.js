import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm, mapStateToProps, mapDispatchToProps } from './SearchForm';

describe('SearchForm', () => {
  describe('SearchForm component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<SearchForm />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

});