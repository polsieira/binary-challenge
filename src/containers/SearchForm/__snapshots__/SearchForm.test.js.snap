// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm SearchForm component should match the snapshot 1`] = `
<div
  className="SearchForm"
>
  <form
    className="search-form"
  >
    <div
      className="input"
    >
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        id="range-slider"
      >
        Distance In Lightyears
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Slider))
        aria-labelledby="range-slider"
        id="distance"
        max={100}
        min={0}
        name="distance"
        onChange={[Function]}
        track="inverted"
        valueLabelDisplay="auto"
      />
    </div>
    <div
      className="input"
    >
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        id="range-slider"
      >
        Comparative Gravity
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Slider))
        aria-labelledby="range-slider"
        id="gravity"
        max={100}
        min={0.0001}
        name="gravity"
        onChange={[Function]}
        track="inverted"
        valueLabelDisplay="auto"
      />
    </div>
    <div
      className="input"
    >
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        id="range-slider"
      >
        Length of Year In Days
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Slider))
        aria-labelledby="range-slider"
        id="year"
        max={2000}
        min={1}
        name="year"
        onChange={[Function]}
        track="inverted"
        valueLabelDisplay="auto"
      />
    </div>
    <button
      className="search-button"
      onClick={[Function]}
      type="button"
    >
      <SearchIcon
        className="search-icon"
      />
    </button>
  </form>
</div>
`;
